pyro/contrib/epidemiology/seir.py

```diff
@@ -0,0 +1,139 @@
+# Copyright Contributors to the Pyro project.
+# SPDX-License-Identifier: Apache-2.0
+
+import torch
+from torch.nn.functional import pad
+
+import pyro
+import pyro.distributions as dist
+from pyro.ops.tensor_utils import convolve
+
+from .compartmental import CompartmentalModel
+
+
+class SimpleSEIRModel(CompartmentalModel):
+    """
+    Susceptible-Exposed-Infected-Recovered model.
+
+    To customize this model we recommend forking and editing this class.
+
+    This is a stochastic discrete-time discrete-state model with four
+    compartments: "S" for susceptible, "E" for exposed, "I" for infected,
+    and "R" for recovered individuals (the recovered individuals are
+    implicit: ``R = population - S - E - I``) with transitions
+    ``S -> E -> I -> R``.
+
+    :param int population: Total ``population = S + E + I + R``.
+    :param float incubation_time: Mean incubation time (duration in state
+        ``E``). Must be greater than 1.
+    :param float recovery_time: Mean recovery time (duration in state
+        ``I``). Must be greater than 1.
+    :param iterable data: Time series of new observed infections, i.e. a
+        Binomial subset of the ``E -> I`` transitions at each time step.
+    """
+
+    def __init__(self, population, incubation_time, recovery_time, data):
+        compartments = ("S", "E", "I")  # R is implicit.
+        duration = len(data)
+        super().__init__(compartments, duration, population)
+
+        assert isinstance(incubation_time, float)
+        assert incubation_time > 1
+        self.incubation_time = incubation_time
+
+        assert isinstance(recovery_time, float)
+        assert recovery_time > 1
+        self.recovery_time = recovery_time
+
+        self.data = data
+
+    series = ("S2E", "E2I", "I2R", "obs")
+    full_mass = [("R0", "rho")]
```

Comment:
should this be passed through `infer` like `num_quant_bins`?

Reply:
Indeed it is already passed like `num_quant_bins`.
https://github.com/pyro-ppl/pyro/blob/a56ecbaf687a6220aa02dc361dbd1d117843d6f6/pyro/contrib/epidemiology/compartmental.py#L247
This class attribute merely sets the default value.